packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,227 @@
+import Box from '~components/Box';
+
+export type TextInputProps = Pick<
+  BaseInputProps,
+  | 'label'
+  | 'labelPosition'
+  | 'necessityIndicator'
+  | 'validationState'
+  | 'helpText'
+  | 'errorText'
+  | 'successText'
+  | 'placeholder'
+  | 'type'
+  | 'defaultValue'
+  | 'name'
+  | 'onChange'
+  | 'onBlur'
+  | 'value'
+  | 'isDisabled'
+  | 'isRequired'
+  | 'prefix'
+  | 'suffix'
+  | 'maxCharacters'
+  | 'autoFocus'
+  | 'keyboardReturnKeyType'
+  | 'autoCompleteSuggestionType'
+> & {
+  /**
+   * Decides whether to render a clear icon button
+   */
+  showClearButton?: boolean;
+  /**
+   * Event handler to handle the onClick event for clear button.
+   */
+  onClearButtonClick?: () => void;
+  /**
+   * Decides whether to show a loading spinner for the input field.
+   */
+  isLoading?: boolean;
+  /**
+   * Icon that will be rendered at the beginning of the input field
+   */
+  icon?: IconComponent;
+};
+
+type TextInputKeyboardAndAutoComplete = Pick<
+  BaseInputProps,
+  'type' | 'keyboardType' | 'keyboardReturnKeyType' | 'autoCompleteSuggestionType'
+>;
+
+const getKeyboardAndAutocompleteProps = ({
+  type,
+  keyboardReturnKeyType,
+  autoCompleteSuggestionType,
+}: TextInputKeyboardAndAutoComplete): TextInputKeyboardAndAutoComplete => {
+  const keyboardAndAutocompleteProps: TextInputKeyboardAndAutoComplete = {
+    type,
+    keyboardType: 'text',
+    keyboardReturnKeyType: 'default',
+    autoCompleteSuggestionType: 'none',
+  };
+
+  if (type === 'text') {
+    keyboardAndAutocompleteProps.keyboardType = 'text';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'default';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  } else if (type === 'telephone') {
+    keyboardAndAutocompleteProps.keyboardType = 'telephone';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType =
+      autoCompleteSuggestionType ?? 'telephone';
+  } else if (type === 'email') {
+    keyboardAndAutocompleteProps.keyboardType = 'email';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'email';
+  } else if (type === 'url') {
+    keyboardAndAutocompleteProps.keyboardType = 'url';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'go';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  } else if (type === 'numeric') {
+    keyboardAndAutocompleteProps.type = 'text';
+    /* the default keyboardType:numeric shows alphanumeric keyboard on iOS but number pad on android. making it type:text and keyboardType:decimal fixes this on all platforms.
+     * source: https://css-tricks.com/everything-you-ever-wanted-to-know-about-keyboardType/#aa-decimal
```

Comment:
Can we use a map instead? Looks a lot cleaner. 

```ts

  const keyboardConfigMap = {
    text: {
      keyboardType: 'text',
      keyboardReturnKeyType: 'default',
      autoCompleteSuggestionType: 'none',
    },
    telephone: {
      keyboardType: 'telephone',
      keyboardReturnKeyType: 'done',
      autoCompleteSuggestionType: 'telephone',
    },
    email: {
      keyboardType: 'email',
      keyboardReturnKeyType: 'done',
      autoCompleteSuggestionType: 'email',
    },
    url: {
      keyboardType: 'url',
      keyboardReturnKeyType: 'go',
      autoCompleteSuggestionType: 'none',
    },
    numeric: {
      keyboardType: 'decimal',
      keyboardReturnKeyType: 'done',
      autoCompleteSuggestionType: 'none',
    },
    search: {
      keyboardType: 'search',
      keyboardReturnKeyType: 'search',
      autoCompleteSuggestionType: 'none',
    },
  } as const;

  const keyboardConfig = keyboardConfigMap[type!];

  keyboardAndAutocompleteProps.keyboardType = keyboardConfig.keyboardType;

  keyboardAndAutocompleteProps.keyboardReturnKeyType =
    keyboardReturnKeyType ?? keyboardConfig.keyboardReturnKeyType;

  keyboardAndAutocompleteProps.autoCompleteSuggestionType =
    autoCompleteSuggestionType ?? keyboardConfig.autoCompleteSuggestionType;

  if (type === 'numeric') {
    /* the default keyboardType:numeric shows alphanumeric keyboard on iOS but number pad on android. making it type:text and keyboardType:decimal fixes this on all platforms.
     * source: https://css-tricks.com/everything-you-ever-wanted-to-know-about-keyboardType/#aa-decimal
     */
    keyboardAndAutocompleteProps.type = 'text';
  }

```

Reply:
fixed in 717f7a1da44ddbaf76edc37e72d60e182d95576b